sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/KeyEncryptionKeyClientBuilder.java

```diff
@@ -114,6 +136,50 @@ public KeyEncryptionKey buildKeyEncryptionKey(String keyId) {
         return Mono.defer(() -> Mono.just(new KeyEncryptionKeyAsyncClient(keyId, pipeline, serviceVersion)));
     }
 
+    /**
+     * Creates a {@link KeyEncryptionKeyAsyncClient} based on options set in the builder.
+     * Every time {@code buildAsyncKeyEncryptionKey(String)} is called, a new instance of
+     * {@link KeyEncryptionKeyAsyncClient} is created.
+     *
+     * <p>If {@link KeyEncryptionKeyClientBuilder#pipeline(HttpPipeline) pipeline} is set, then the {@code pipeline}
+     * and {@code keyId} are used to create the {@link KeyEncryptionKeyAsyncClient async client}.
+     * All other builder settings are ignored. If {@code pipeline} is not set, then
+     * ({@link KeyEncryptionKeyClientBuilder#credential(TokenCredential) jsonWebKey vault credential} and
+     * {@code keyId} are required to build the {@link KeyEncryptionKeyAsyncClient async client}.</p>
+     *
+     * @param key the key to be used for crypto operations.
+     *
+     * @return A {@link KeyEncryptionKeyAsyncClient} with the options set from the builder.
+     * @throws IllegalStateException If {@link KeyEncryptionKeyClientBuilder#credential(TokenCredential)} or
+     * {@code key} have not been set.
+     */
+    public Mono<? extends AsyncKeyEncryptionKey> buildAsyncKeyEncryptionKey(KeyVaultKey key) {
+        builder.key(key);
+        if (key == null) {
+            throw logger.logExceptionAsError(new IllegalStateException(
+                "Key is required to create key encryption key async client"));
+        }
+
+        if (key.getKey() == null) {
+            throw logger.logExceptionAsError(new IllegalStateException(
+                "Json Web key is required to create key encryption key async client"));
```

Comment:
These exceptions do not clearly state what the user should do to resolve the issue. More specificity means less confusion for end users.

Reply:
Updated the messages.